### PR TITLE
adapt bounds of polygon in test

### PR DESCRIPTION
**Pull request by jdroenner**
- [ ] I added an entry to [`CHANGELOG.md`](CHANGELOG.md) if knowledge of this change could be valuable to users.

---

Here is a brief summary of what I did:
i changed the polygon bounds to not wrap into the next tile

### CODE REVIEW

**Comment by jdroenner** [comment]
[Polygon Raster Join Test mehr robust machen..](https://trello.com/c/APbqZhJ3/1173-polygon-raster-join-test-mehr-robust-machen)

File: datatypes/src/primitives/spatial_partition.rs [hunk 1]
```diff
@@ -70,12 +70,21 @@ impl SpatialPartition2D {
     /// Create a partition from a bbox by snapping to the next pixel
     /// The resulting partition is not equivalent to the bbox but contains it
     pub fn with_bbox_and_resolution(bbox: BoundingBox2D, resolution: SpatialResolution) -> Self {
+        const EPSILON: f64 = 0.000_001;
```
Comment (michaelmattig) [inline datatypes/src/primitives/spatial_partition.rs:73]: maybe add a todo to get rid of this epsilon and just snap the the next pixel, once spatial partitions are reworked.

File: datatypes/src/primitives/spatial_partition.rs [hunk 2]
```diff
@@ -70,12 +70,21 @@ impl SpatialPartition2D {
...
+        let epsilon_y = resolution.y * EPSILON;
+
         Self {
             upper_left_coordinate: bbox.upper_left(),
+
             lower_right_coordinate: Coordinate2D {
-                x: (lr.x / resolution.x).ceil() * resolution.x,
-                y: (lr.y / resolution.y).ceil() * resolution.y,
+                x: ((lr.x + epsilon_x) / resolution.x).ceil() * resolution.x,
+                y: ((lr.y + epsilon_y) / resolution.y).ceil() * resolution.y,
```
Comment (michaelmattig) [inline datatypes/src/primitives/spatial_partition.rs:87]: shouldn't this bis - eplyson_y?

File: datatypes/src/primitives/spatial_partition.rs [hunk 3]
```diff
@@ -70,12 +70,21 @@ impl SpatialPartition2D {
     /// Create a partition from a bbox by snapping to the next pixel
     /// The resulting partition is not equivalent to the bbox but contains it
     pub fn with_bbox_and_resolution(bbox: BoundingBox2D, resolution: SpatialResolution) -> Self {
+        const EPSILON: f64 = 0.000_001;
+
         let lr = bbox.lower_right();
+
+        // add an epsilon to the lower right coordinate to make sure that all pixels are included
+
+        let epsilon_x = resolution.x * EPSILON;
```
[the remark this answers is not in the document]
Comment (jdroenner) [reply datatypes/src/primitives/spatial_partition.rs]: i guess if we hit the float min we have other issues

File: datatypes/src/primitives/spatial_partition.rs (hunk 2 above)
Comment (jdroenner) [reply datatypes/src/primitives/spatial_partition.rs]: yes. i reworked the whole thing and adapted tests...

[bdce194](https://github.com/geo-engine/geoengine/pull/712/commits/bdce194e9986dca3733883c4931520219fc92aeb)

**Review by michaelmattig — DISMISSED** [review]

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**